### PR TITLE
Add a curated output flag for inventory

### LINKED ISSUE #306: Provide context when viewing broker inventory of what each machine is
I would love to have a way when viewing `broker inventory` to be able to define and see context of the machines in order to remember what each was. For example, I often use the `--nick` feature to spin up machines based on a "profile". That nick is typically descriptive, e.g. `satellite-6.15`. When I am staring at a list of multiple machines in inventory I have to guess which was which, accessing each and reminding myself which is which to find the one I need. This would be great to extend to the operating system running as well.

Note: I am not saying that it's the nick that has to be used, could be some description property or something else entirely that provides that context.

### PR DESCRIPTION
**Pull request by JacobCallahan**
This provides a styled middle ground between the default minimal view and the much more verbose detailed view.
This also give us an opportunity to explore different output styling with rich.

fixes #306

### CODE REVIEW

**Comment by JacobCallahan** [comment]
And this is what it looks like

Screenshot from 2024-09-06 16-52-46

**Review by ehelms — APPROVED** [review]
I'd love it if this was the default view, but i'll take it either way.

File: broker/commands.py [hunk 1]
```diff
@@ -282,9 +286,25 @@ def inventory(details, sync, filter):
...
+    helpers.emit({"inventory": inventory})
+    if curated:
+        console = Console()
+        table = Table(title="Host Inventory")
+
+        table.add_column("Id", justify="left", style="cyan", no_wrap=True)
+        table.add_column("Host", justify="left", style="magenta")
+        table.add_column("Provider", justify="left", style="green")
+        table.add_column("Action", justify="left", style="yellow")
+        table.add_column("OS", justify="left", style="blue")
```
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply broker/commands.py:298]: I'd like to keep this table small, and not replicate the details view. So for now, I do want to stick with these columns, but look into ways to make it customizable in the future.

**Comment by JacobCallahan** [comment]
@ehelms there's a good chance that I'll make this the default view in the future, but since this is coming out in a Z release, I don't want to break existing workflows.

**Pull request merged**